chain/network/src/peer_manager/db_types.rs

```diff
@@ -0,0 +1,55 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use chrono::{DateTime, Utc};
+use near_network_primitives::types::{PeerInfo, ReasonForBan};
+use near_primitives::time::Clock;
+use near_primitives::utils::{from_timestamp, to_timestamp};
+
+/// Status of the known peers.
+#[derive(BorshSerialize, BorshDeserialize, Eq, PartialEq, Debug, Clone)]
+pub enum KnownPeerStatus {
+    Unknown,
+    NotConnected,
+    Connected,
+    Banned(ReasonForBan, u64),
+}
+
+impl KnownPeerStatus {
+    pub fn is_banned(&self) -> bool {
+        match self {
+            KnownPeerStatus::Banned(_, _) => true,
+            _ => false,
+        }
+    }
+}
+
+/// Information node stores about known peers.
+#[derive(BorshSerialize, BorshDeserialize, Debug, Clone)]
+pub struct KnownPeerState {
+    pub peer_info: PeerInfo,
+    pub status: KnownPeerStatus,
+    pub first_seen: u64,
+    pub last_seen: u64,
+}
+
+impl KnownPeerState {
+    pub(crate) fn new(peer_info: PeerInfo) -> Self {
+        KnownPeerState {
+            peer_info,
+            status: KnownPeerStatus::Unknown,
+            first_seen: to_timestamp(Clock::utc()),
+            last_seen: to_timestamp(Clock::utc()),
+        }
+    }
+
+    pub(crate) fn last_seen(&self) -> DateTime<Utc> {
```

Comment:
would be nice to get rid of chrono dependency here while we are at it, and if that's not too much bigger diff.

Reply:
@matklad See https://github.com/near/nearcore/pull/5556#pullrequestreview-831602436
It has been stuck in review queue for 2 weeks.